### PR TITLE
Java enum generation

### PR DESCRIPTION
**Pull request by eugene-yao-zocdoc**
@christopher-taormina-zocdoc @chris-smith-zocdoc @ricky-hartmann-zocdoc 

In scala, our enums are better off as java. I've tested the plinth code using these code gen functions and parity has been achieved.

### CODE REVIEW

File: java/writer.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+func (self *codeWriter) End() {
+	self.indentation -= 1
+	self.Write("}")
+	self.Eol()
+}
+
+func (self *codeWriter) Eol() {
+	if self.newLine {
+		self.newLine = false
```
Comment (eugene-yao-zocdoc) [inline java/writer.go:39]: Scala and I'm guessing Typescript need this fix to prevent empty lines from having spaces at the beginning.

**Comment by christopher-taormina-zocdoc** [comment]
No tests?

File: java/writer.go (hunk 1 above)
Comment (chris-smith-zocdoc) [reply java/writer.go:39]: This occurs when you have a double newline? Or some other scenario?

**Comment by eugene-yao-zocdoc** [comment]
> No tests?

Nope.

**Comment by christopher-taormina-zocdoc** [comment]
> > No tests?
> 
> Nope.

They're pretty easy https://github.com/Zocdoc/gopoetry/blob/master/scala/method_test.go

File: java/writer.go (hunk 1 above)
Comment (eugene-yao-zocdoc) [reply java/writer.go:39]: @chris-smith-zocdoc yea double newline

**Pull request merged**